crates/encoding/src/glyph_cache.rs

```diff
@@ -145,3 +157,111 @@ impl OutlinePen for BezPathPen {
         self.0.close_path();
     }
 }
+
+/// We keep this small to enable a simple LRU cache with a linear
+/// search. Regenerating hinting data is low to medium cost so it's fine
+/// to redo it occassionally.
+const MAX_CACHED_HINT_INSTANCES: usize = 8;
+
+pub struct HintKey<'a> {
+    font_id: u64,
+    font_index: u32,
+    outlines: &'a OutlineGlyphCollection<'a>,
+    size: Size,
+    coords: &'a [NormalizedCoord],
+}
+
+impl<'a> HintKey<'a> {
+    fn new(
+        outlines: &'a OutlineGlyphCollection<'a>,
+        glyph_key: &GlyphKey,
+        size: f32,
+        coords: &'a [NormalizedCoord],
+    ) -> Self {
+        Self {
+            font_id: glyph_key.font_id,
+            font_index: glyph_key.font_index,
+            outlines,
+            size: Size::new(size),
+            coords,
+        }
+    }
+
+    fn instance(&self) -> Option<HintingInstance> {
+        HintingInstance::new(self.outlines, self.size, self.coords, HINTING_MODE).ok()
+    }
+}
+
+const HINTING_MODE: HintingMode = HintingMode::Smooth {
+    lcd_subpixel: Some(LcdLayout::Horizontal),
```

Comment:
I'm guessing this will need to be wired through at some point?

Reply:
That’s an open question. The skrifa hinter supports a lot of modes because we have a requirement of matching FreeType. There’s an argument to be made that vello should always just use a modern hinting mode but that’s open to discussion. Ultimately, most fonts these days have hints generated by ttfautohint which ignores these things. Only the MS ClearType fonts really take advantage of them and we can potentially choose a sane default for those.